From 1ae3d410cac60bbf78682df4a0689d56fb3b80d4 Mon Sep 17 00:00:00 2001
From: Peter Delevoryas <pdel@meta.com>
Date: Sun, 18 Dec 2022 14:41:31 -0800
Subject: [PATCH 527/532] hw/arm/aspeed: Don't initialize fby35-bmc GPIO's

Initializing the GPIO's triggers slot sensor cache initialization in the
boot sequence, which is extremely slow and annoying and doesn't succeed.

Until we update the BMC's bic-cached setup to be less annoying, it's
better to just not initialize these GPIO's.

Signed-off-by: Peter Delevoryas <pdel@meta.com>
---
 hw/arm/aspeed.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/hw/arm/aspeed.c b/hw/arm/aspeed.c
index 32f0a770f7..40671c9f93 100644
--- a/hw/arm/aspeed.c
+++ b/hw/arm/aspeed.c
@@ -1572,6 +1572,7 @@ static void aspeed_machine_bletchley_class_init(ObjectClass *oc, void *data)
     aspeed_machine_class_init_cpus_defaults(mc);
 }
 
+#if 0
 static void fby35_reset(MachineState *state, ShutdownCause reason)
 {
     AspeedMachineState *bmc = ASPEED_MACHINE(state);
@@ -1579,6 +1580,7 @@ static void fby35_reset(MachineState *state, ShutdownCause reason)
     qemu_devices_reset(reason);
     fby35_bmc_reset_gpios(bmc->soc);
 }
+#endif
 
 static void aspeed_machine_fby35_class_init(ObjectClass *oc, void *data)
 {
@@ -1586,7 +1588,7 @@ static void aspeed_machine_fby35_class_init(ObjectClass *oc, void *data)
     AspeedMachineClass *amc = ASPEED_MACHINE_CLASS(oc);
 
     mc->desc       = "Facebook fby35 BMC (Cortex-A7)";
-    mc->reset      = fby35_reset;
+    //mc->reset      = fby35_reset;
     amc->fmc_model = "n25q00";
     amc->num_cs    = 2;
     amc->macs_mask = ASPEED_MAC3_ON;
-- 
2.44.2

